### PR TITLE
DATAGRAPH-1393 - Add support for OGM's CompositeAttributeConverter.

### PR DESCRIPTION
**Pull request by michael-simons**
Like OGM, but better.

### CODE REVIEW

File: src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentProperty.java [hunk 1]
```diff
@@ -182,7 +183,7 @@ public boolean isAssociation() {
 
 	@Override
 	public boolean isEntity() {
-		return super.isEntity() && isAssociation() || (super.isEntity() && isEntityInRelationshipWithProperties());
+		return super.isEntity() && isAssociation() || (super.isEntity() && isEntityInRelationshipWithProperties() && !isAnnotationPresent(CompositeProperty.class));
```
Comment (meistermeier) [inline src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentProperty.java:186]:
```suggestion
		return super.isEntity() && isAssociation() || (super.isEntity() && isEntityInRelationshipWithProperties() && !isComposite());
```

Comment (meistermeier) [reply src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentProperty.java:186]: I prefer a central access to the annotation more than having the same (inversed) logic written again.

**Review by meistermeier — APPROVED** [review]
LGTM. As always just nitpicking.

File: src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentProperty.java (hunk 1 above)
Comment (michael-simons) [reply src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentProperty.java]: Me too. Thanks.

**Comment by michael-simons** [comment]
Merged as 7f75f13ef.

**Pull request closed without merging**